pulp_ansible/app/models.py

```diff
@@ -45,44 +43,67 @@ class Meta:
+        homepage (models.URLField): The URL to the homepage of the collection/project.
+        issues (models.URLField): The URL to the collection issue tracker.
+        license (psql_fields.ArrayField): A list of licenses for content inside of a collection.
+        license_file (models.CharField): The path to the license file for the collection.
+        name (models.CharField): The name of the collection.
+        namespace (models.CharField): The namespace of the collection.
+        readme (models.TextField): The path to the Markdown (.md) readme file.
+        repository (models.URLField): The URL of the originating SCM repository.
+        version (models.CharField): The version of the collection.
 
     Relations:
 
-        collection: Reference to a collection model.
+        collection (models.ForeignKey): Reference to a collection model.
+        dependencies (models.ManyToManyField): An asymmetric reference to other CollectionVersion
+            objects.
+        tag (models.ManyToManyField): A symmetric reference to the Tag objects.
     """
 
     TYPE = "collection_version"
 
+    authors = psql_fields.ArrayField(models.CharField(max_length=64), editable=False)
+    contents = psql_fields.JSONField(default=list, editable=False)
+    description = models.TextField(default="", blank=True, editable=False)
+    docs_blob = psql_fields.JSONField(default=dict, editable=False)
+    documentation = models.URLField(default="", blank=True, max_length=128, editable=False)
+    homepage = models.URLField(default="", blank=True, max_length=128, editable=False)
+    issues = models.URLField(default="", blank=True, max_length=128, editable=False)
+    license = psql_fields.ArrayField(models.CharField(max_length=32), editable=False)
+    license_file = models.CharField(default="", blank=True, max_length=32, editable=False)
+    name = models.CharField(max_length=32, editable=False)
+    namespace = models.CharField(max_length=32, editable=False)
+    readme = models.TextField(default="", blank=True, editable=False)
+    repository = models.URLField(default="", blank=True, max_length=128, editable=False)
+    version = models.CharField(max_length=32, editable=False)
+
     collection = models.ForeignKey(
         Collection, on_delete=models.CASCADE, related_name="versions", editable=False
     )
-    version = models.CharField(max_length=128, editable=False)
-
-    quality_score = models.FloatField(null=True, editable=False)
-    metadata = psql_fields.JSONField(default=dict, editable=False)
-    contents = psql_fields.JSONField(default=list, editable=False)
-
-    @property
-    def namespace(self):
-        """Returns collection namespace name."""
-        return self.collection.namespace
-
-    @property
-    def name(self):
-        """Returns collection name."""
-        return self.collection.name
+    dependencies = models.ManyToManyField("self", symmetrical=False, editable=False)
```

Comment:
Dependencies shouldn't be stored as many-to-many. A dependency version requirement is a version range specifier for example `> 1.2.3`, which means the following:

1. A collection may have more than one version of a specific collection it depends on. (See my comment below).
2. This relationship does not store information about version ranges provided by metadata.
3. Maintaining such dependency list requires checking for reverse dependencies for entire database. Let's say a collection `A` has a dependency on a collection `B = ">1.2.3"`. Collection `A` dependencies list has to be updated each time new version of a collection `B` is imported to pulp.

Reply:
I agree we should just store the data since it's an expression of version specifier not a specific named one. Let me rework this and then let's see what you think.